crates/router/tests/connectors/rapyd.rs

```diff
@@ -0,0 +1,232 @@
+        flow: PhantomData,
+        merchant_id: "rapyd".to_string(),
+        connector: "rapyd".to_string(),
+        payment_id: uuid::Uuid::new_v4().to_string(),
+        attempt_id: None,
+        status: enums::AttemptStatus::default(),
+        router_return_url: None,
+        auth_type: enums::AuthenticationType::NoThreeDs,
+        payment_method: enums::PaymentMethodType::Card,
+        connector_auth_type: auth.into(),
+        description: Some("This is a test".to_string()),
+        return_url: None,
+        request: types::PaymentsAuthorizeData {
+            amount: 100,
+            currency: enums::Currency::USD,
+            payment_method_data: types::api::PaymentMethod::Card(api::CCard {
+                card_number: "4111111111111111".to_string().into(),
+                card_exp_month: "02".to_string().into(),
+                card_exp_year: "24".to_string().into(),
+                card_holder_name: "John Doe".to_string().into(),
+                card_cvc: "123".to_string().into(),
+            }),
+            confirm: true,
+            statement_descriptor_suffix: None,
+            setup_future_usage: None,
+            mandate_id: None,
+            off_session: None,
+            setup_mandate_details: None,
+            capture_method: None,
+            browser_info: None,
+            order_details: None,
+            email: None,
+        },
+        response: Err(types::ErrorResponse::default()),
+        payment_method_id: None,
+        address: PaymentAddress::default(),
+        connector_meta_data: None,
+        amount_captured: None,
+    }
+}
+
+fn construct_refund_router_data<F>() -> types::RefundsRouterData<F> {
+    let auth = ConnectorAuthentication::new()
+        .rapyd
+        .expect("Missing Rapyd connector authentication configuration");
+
+    types::RouterData {
+        flow: PhantomData,
+        connector_meta_data: None,
+        merchant_id: "rapyd".to_string(),
+        connector: "rapyd".to_string(),
+        payment_id: uuid::Uuid::new_v4().to_string(),
+        attempt_id: None,
+        status: enums::AttemptStatus::default(),
+        router_return_url: None,
+        payment_method: enums::PaymentMethodType::Card,
+        auth_type: enums::AuthenticationType::NoThreeDs,
+        connector_auth_type: auth.into(),
+        description: Some("This is a test".to_string()),
+        return_url: None,
+        request: types::RefundsData {
+            amount: 100,
+            currency: enums::Currency::USD,
+            refund_id: uuid::Uuid::new_v4().to_string(),
+            connector_transaction_id: String::new(),
+            refund_amount: 10,
+            connector_metadata: None,
+            reason: None,
+        },
+        response: Err(types::ErrorResponse::default()),
+        payment_method_id: None,
+        address: PaymentAddress::default(),
+        amount_captured: None,
+    }
+}
+
+#[actix_web::test]
+async fn test_rapyd_payment_success() {
```

Comment:
we can implement the test like `Shift4` connector because there is common handling functions and traits. Which can auto generated also.

Reply:
sure